ocs_ci/utility/utils.py

```diff
@@ -2313,14 +2316,61 @@ def ceph_health_check_base(namespace=None):
     """
     namespace = namespace or config.ENV_DATA["cluster_namespace"]
     health = run_ceph_health_cmd(namespace)
-
     if health.strip() == "HEALTH_OK":
         log.info("Ceph cluster health is HEALTH_OK.")
         return True
     else:
         raise CephHealthException(f"Ceph cluster health is not OK. Health: {health}")
 
 
+def ceph_health_check_internal_and_external(cluster_platform):
```

Comment:
I would tackle it differently.

Have separate function to perform only external cluster health check.
Add parameter `cluster_type`  (allowed values: internal_and_external, internal, external)to those functions:
`ceph_health_check_base` and `ceph_health_check`
Set default value for  `cluster_type="internal_and_external"`

Update those two function to tackle in the loop health check on internal, external, both based on the parameter.
Of course in case we don't run on multicluster type of deployment we will skip the external part.

**The reason:**
We already call many times in our tests the ceph health check function: `ceph_health_check`.
Where we are making sure the ceph is in healthy state - I would still except that in multicluster deployment, after any test - both internal and external cluster should be in healthy state.

I see you introduced some new exceptions, but I would do it this way if you want to introduce new CephHealthExceptions I would create:
`InternalCephHealthException` and `ExternalCephHealthException`
where both should inherit from original `CephHealthException` .

In case we already rely on ond except such exception `CephHealthException` in the code, then the generic
```python
except CephHealthException as ex:
```
Will cover also new exceptions which will tell us more precise where the health issue occurred.

Reply:
**Have separate function to perform only external cluster health check** - The function support namespace which can serve us for internal or external, openshift-storage for internal and openshift-storage-extended for external. So no need for separate function. Just need a parameter for multi or act on config.DEPLOYMENT key.
In general I understand the need for before and after test. I'll will implement the function but leave me space to decide about the parameters. I'll do it in a manner that we will have the minimal changes to tests and framework.

Reply:
just summarizing the call we had.

- We wil add new parameter to ceph_health_check and ceph_health_check_base function to parametrize if it should do internal, external or both. (We can update namespace parameter docs - that this parameter is applicable only for internal mode deployment.) 
- This new parameter will be by default `internal` (Because all current use cases were only care of the internal mode - or regular external mode)
- All calls of ceph_health_check function in ceph_health_checker fixture: https://github.com/red-hat-storage/ocs-ci/blob/11cf368e1a02cb232b8d8dbf39c6e781638e44b4/tests/conftest.py#L1620 we will by default have `internal` , but in case it's multicluster deployment - we will check health of `both`.

Reply:
Agree to all but **we will by default have internal , but in case it's multicluster deployment - we will check health of both** there will be no default because there will a variable in the ceph_health when calling the  function the parameter that will be decided upon all conditions checks.

Reply:
In the function definition there will be default to "internal".